diff --git a/tasks/presym.rake b/tasks/presym.rake
index 7a9262fad..ed0fea62a 100644
--- a/tasks/presym.rake
+++ b/tasks/presym.rake
@@ -2,7 +2,8 @@ all_prerequisites = ->(task_name, prereqs) do
   Rake::Task[task_name].prerequisites.each do |prereq_name|
     next if prereqs[prereq_name]
     prereqs[prereq_name] = true
-    all_prerequisites.(Rake::Task[prereq_name].name, prereqs)
+    task = Rake.application.lookup(prereq_name)
+    all_prerequisites.(task.name, prereqs) if task
   end
 end
 
@@ -65,7 +66,9 @@ MRuby.each_target do |build|
     next unless File.extname(prereq) == build.exts.object
     next unless prereq.start_with?(build_dir)
     next if mrbc_build_dir && prereq.start_with?(mrbc_build_dir)
-    file prereq => presym_proxy
+    if Rake.application.lookup(prereq)
+      Rake::Task[prereq].enhance([presym_proxy])
+    end
   end
 
   task gensym: presym.list_path
